include/platform/clinician_input.h

```diff
@@ -13,3 +13,13 @@
  *
  */
 status_t get_respiratory_rate(uint32_t * RR);
+
+/*
+ * Updates <selected inspiratory time> variable insp_time 
+ * when a valid value was read from the clinician inputs
+ * 
+ * @return STATUS_OK if <selected inspiratory time> is valid 
+ *         STATUS_ERR if the r<selected inspiratory time> is invalid
+ *
+ */
+status_t get_selected_inspiratory_time(uint32_t * insp_time);
```

Comment:
if this is true just a getter from a memory location, than you can revise this to directly return the value, rather than using the status/pointer pattern. 

The status/pointer approach makes sense when the operation *could* fail due to some dependency on hardware (or whatever else). However, in this case, if it is just a getter, then this operation should always succeed.

Reply:
This is a platform function. They all follow the same style. moreover We need the platform team to dedcide how the knobs will work. This is a work around. we will see in the future how the knobs really work